ui/src/pages/landing/Landing.tsx

```diff
@@ -257,11 +261,27 @@ const Landing = () => {
       .reduce((sum, deposit) => sum + +deposit.payload.asset.quantity, 0)
   );
 
+  useEffect(() => {
+    console.log('state change');
+    const provider =
+      identities.find(i => i.payload.legalName === dialogState?.provider)?.payload.customer || '';
+    const filteredFields: Fields = Object.fromEntries(
+      Object.entries(filterFields).map(([k, v]) => [k, v(provider)])
+    );
```

Comment:
What exactly does this `Object.fromEntries(Object.entries(...)....)` stuff do? I think I saw another similar thing down below too

Reply:
I suppose to be more precise with my question, what function is `v`?

Reply:
I could definitely name it better than `v`, but it's a function `(provider: Party) => Field` that's meant to run every time the provider changes to get a filtered version of the list items (although you could  technically do anything you want, so maybe `filterField` is misnamed as well)...so an example of what `v` might be is:
```typescript
    provider => {
      return {
        label,
        type: 'selection',
        items: assetSettlementRules
          .filter(
            ar => ar.payload.observers.map.has(provider) || ar.payload.account.provider == provider
          )
          .map(ar => ar.payload.account.id.label),
      };
```

Reply:
gotcha, thanks!

Reply:
Might be clearer like this: 
```typescript
const filteredFields: Fields = _.mapValues(fieldFromProviders, createFieldFn => createFieldFn(provider));
```

Reply:
yeah! that's a lot better actually

Reply:
and +1 for lodash, there are probably more opportunities to use it more often... so many useful little utilities in there

Reply:
for sure!

Reply:
I changed it to use the lodash function and these types (which I was able to reuse in the notifications side of it):
```typescript
export type FieldCallback<T> = (arg: T) => Field;
export type FieldCallbacks<T> = Record<string, FieldCallback<T>>;
```

I'm glad you asked for clarification, this  all makes it much cleaner!